Feature/user management

OgawaManami opened the pull request:
【gyazo】
・必要な情報を適切に入力して「会員登録」ボタンを押すと、ユーザーの新規登録ができる動画
https://gyazo.com/47cdc2bdea3be26f9853b228a21d3e4a

・入力に問題のある状態で「会員登録」ボタンが押された場合、ユーザーの新規登録はできず、エラーメッセージが出力される動画
https://gyazo.com/f4c0ddce4b7b1a211ad8cb1ab666c4d6

・必要な情報を適切に入力して「ログイン」ボタンを押すと、ユーザーのログインができる動画
https://gyazo.com/e90f963d3c5a3835ccd7d6e44c23e013

・入力に問題のある状態で「ログイン」ボタンが押された場合、ユーザーのログインはできず、エラーメッセージが出力される動画
https://gyazo.com/2f89258ccb47928f27b20169823d0071

・ログアウトに成功する動画
https://gyazo.com/55ecc36281c88a8302896f03013e6498

・テストに成功する画像
すみません、長いので２つあります
＜前半＞
https://gyazo.com/fe86f19d67e78f82ea3c2b3f46e39fd6
＜後半＞
https://gyazo.com/e5162c7d062840a46fe245fc36280424
＜動画＞
https://gyazo.com/d208049b79d0ca353824e0c6008da7db

OgawaManami committed:
deviseのインストール/usersテーブルの作成

# What
・deviseのインストール
・ usersテーブルに追加するカラムをマイグレイトファイルに記述
・マイグレーションの実行

# Why
ユーザー管理機能実装のため

OgawaManami committed:
ビューファイル修正/バリデーション設定

# What
・【修正】
app/views/registrations/new.html.erb

・【バリデーション設定】
app/models/user.rd

# Why
ユーザー管理機能実装のため

OgawaManami committed:
ActiveHash導入

# What
・Gemの追加
・prefecture.rbファイル作成・記述

# Why
都道府県情報をActiveHashを用いて保存するように実装するため

OgawaManami committed:
RSpec導入

# What
・RSpecインストール
・.rspecに設定を追加
・Userモデルのテストファイルを生成

# Why
テストコード記述のため

OgawaManami committed:
モデル単体テストコード

# What
モデル単体テストコードの記述
・spec/models/user_spec.rb
・spec/factories/users.rb

# Why
ユーザー管理機能実装のため

OgawaManami committed:
新規登録・ログイン機能実装

# What
・新規登録画面実装
・ログイン画面実装

# Why
ユーザー管理機能実装のため

div-reviewer commented on app/views/shared/_header.html.erb:
```diff
@@ -15,12 +15,17 @@
-    </ul>
+    <% if user_signed_in? %>
+      <div class="nav__right">
+      <%= link_to current_user.nickname, destroy_user_session_path, class: "user-nickname" %>
```
furimaの必須機能にはユーザー詳細機能が入っていないため、パスは仮置き（ "#"）のままで大丈夫です。

OgawaManami replied:
もしかして、ログアウトも同様でしょうか…
20行目・21行目を配布のfurima素材に修正しました
もし解釈が違っていたら恐れ入りますが教えていただけますと幸いです！

div-reviewer commented on db/migrate/20230923092556_devise_create_users.rb:
```diff
@@ -0,0 +1,50 @@
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :nickname,           null: false
+      t.string :email,              null: false, unique: true
```
◎unique:trueの記述を削除しましょう。

■理由
deviseがデフォルトで付与しているためです。

div-reviewer commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,17 @@
+
+    nickname { Faker::Name.name }
+    email { Faker::Internet.free_email }
+    password { Faker::Internet.password(min_length: 6) }
```
◎passwordカラムを修正しましょう。
現状、必ずしも半角英数混合になるかどうかが不明となる記述です。
(Fakerのみの場合、英字のみのパスワードが生成されてしまうことがあるためです。)
確実に半角英数混合の値を生成するため、『+』の記号を使用し、文字列を組み合わせることで期待する値を生成する内容に修正していきましょう。

div-reviewer reviewed (commented):
OgawaManami様、学習お疲れ様です！
ユーザー管理機能の初回レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

完成度高く実装されているかと思われます。今までの学習がバッチリ成果として出ていますね。大変素晴らしいです。

修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！**
どうぞよろしくおねがいします。

OgawaManami committed:
ユーザー管理機能実装 修正1

# What
＜プルリクエストに対するコメントの内容修正＞
・app/views/shared/_header.html.erb
・db/migrate/20230923092556_devise_create_users.rb
・spec/factories/users.rb

ShotaOtake reviewed (commented):
レビュー依頼を頂きありがとうございます！
お送り頂いた内容を確認致しました。

今回【LGTM】とさせて頂きます。
実装お疲れ様でした！

```
※現在のブランチはマージ後、ローカルのmasterブランチに忘れずにプルしましょう！
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
```

引き続き実装頑張っていきましょう！
応援しております！
